testplan/testing/multitest/base.py

```diff
@@ -802,9 +806,25 @@ def _execute_step(self, step, *args, **kwargs):
         with self.report.logged_exceptions():
```

Comment:
For multitest, all exceptions happened during executing steps are captured in self.report right?
Even for environment start/stop, the exception first goes to self.resources.start_exceptions/stop_exceptions, are collected info self.report in post_step_call?
IMO, we can look at self.report.status == error as a simpler indicator of whether we shall invoke error_handler.

Also when to invoke the error_handler is something we shall consider - right after the exceptioned step or as a final safe net after other major steps. For example, if after_start exceptioned, shall we call error_handler immediately or after stop? 

The existing code is controlling the runnable execution flow with step definition, thus a if-else branch inside _execution_step is breaking style. The steps mechanism meant to avoid such spaghetti code. You can move this to post_step_call.

Another thing is, I feel this new config option and related logic shall be promoted to Test class.